request.go

```diff
@@ -93,22 +96,41 @@ func NewRequest(config Config, method, path string, pathargs ...interface{}) *Re
 		return nil
 	})
 	if err != nil {
-		chain.fail(err.Error())
+		chain.fail(NewErrorFailure(err))
 	}
 
 	hr, err := config.RequestFactory.NewRequest(method, config.BaseURL, nil)
 	if err != nil {
-		chain.fail(err.Error())
+		chain.fail(NewErrorFailure(err))
 	}
 
 	return &Request{
-		config: config,
-		chain:  chain,
-		path:   path,
-		http:   hr,
+		config:  config,
+		chain:   chain,
+		path:    path,
+		http:    hr,
+		context: placeholderCtx,
 	}
 }
 
+// WithContext is NOT go's standard context.Context interface. It is a concrete *Context pointer from httpexpect
+// that holds data to be provided to an AssertionHandler (also Reporter and Formatter, when using the default one).
+//
+// Do NOT use this method yourself unless you know what you are doing: Expect.Request handles the call and passes
+// the current Expect's Context to the Request as needed.
+//
+// WithContext replaces the existing context with the given one.
+//
+// Providing a nil Context will panic.
+func (r *Request) WithContext(ctx *Context) *Request {
```

Comment:
Huh, in master we already have this: https://github.com/gavv/httpexpect/blob/master/request.go#L453-L465

But do you think we really need this method, at all?

I though about Context as something that httpexpect maintains and provides to the user (to AssertionFormatter implementation). Through Context, httpexpect describes its state (current request, etc) at the moment of failure.

But what is the use case when *the user* should provide us with context?

Reply:
ouh. didn't see that. maybe it was changed in the `gavv:context` branch?

Reply:
I think context branch was forked from master before this change was added to master. We'll need to rebase context on master eventually.